src/storage/src/storage_state.rs

```diff
@@ -116,7 +116,7 @@ impl<'a, A: Allocate, B: StorageCapture> ActiveStorageState<'a, A, B> {
             StorageCommand::CreateSources(sources) => {
                 for source in sources {
                     match &source.desc.connector {
-                        SourceConnector::Local { .. } => {
+                        SourceConnector::Local { .. } | SourceConnector::Log => {
                             self.storage_state.table_state.insert(
```

Comment:
This probably should just panic if it sees a SourceConnector::Log, as you can’t create a log source manually. We don’t presently ever send a CreateSources command with a log source, so it works out ok now, but given that we now have the language to talk about log and table sources separately, might as well make the distinction in behavior explicit!

Reply:
We actually do send `CreateSources` commands for log sources. It happens during bootstrapping the coordinator, where it loads all sources from the catalog, including the built-in ones, and announces them to storage: https://github.com/MaterializeInc/materialize/blob/main/src/coord/src/coord.rs#L558-L582. So if I change that code to panic, materialized isn't able to start up anymore.

I assume we could adjust the coordinator to not send these `CreateSources` commands in the first place, but I'd rather wait for @petrosagg's separation PR to land, since it touches that code anyway.

Reply:
Oh, sigh. I guess we’ve just been incorrectly creating tables for all these logging sources? Can you file a follow up issue and assign to me?